lnbits/wallets/cliche.py

```diff
@@ -0,0 +1,134 @@
+import asyncio
+import json
+from os import getenv
+from typing import AsyncGenerator, Dict, Optional
+
+import httpx
+from loguru import logger
+from websocket import create_connection
+
+from .base import (
+    InvoiceResponse,
+    PaymentResponse,
+    PaymentStatus,
+    StatusResponse,
+    Wallet,
+)
+
+
+class ClicheWallet(Wallet):
+    """https://github.com/fiatjaf/cliche"""
+
+    def __init__(self):
+        self.endpoint = getenv("CLICHE_ENDPOINT")
+
+    async def status(self) -> StatusResponse:
+        try:
+            ws = create_connection(self.endpoint)
+            ws.send("get-info")
+            r = ws.recv()
+        except Exception as exc:
+            return StatusResponse(
+                f"Failed to connect to {self.endpoint} due to: {exc}", 0
+            )
+        try:
+            data = json.loads(r)
+        except:
+            return StatusResponse(
+                f"Failed to connect to {self.endpoint}, got: '{r.text[:200]}...'", 0
+            )
+
+        return StatusResponse(None, data["result"]["wallets"][0]["balance"])
+
+    async def create_invoice(
+        self,
+        amount: int,
+        memo: Optional[str] = None,
+        description_hash: Optional[bytes] = None,
+    ) -> InvoiceResponse:
+        if description_hash:
+            ws = create_connection(self.endpoint)
+            ws.send(
+                f"create-invoice --msatoshi {amount*1000} --description_hash {description_hash}"
+            )
+            r = ws.recv()
+        else:
+            ws = create_connection(self.endpoint)
+            ws.send(f"create-invoice --msatoshi {amount*1000} --description {memo}")
+            r = ws.recv()
+        data = json.loads(r)
+        checking_id = None
+        payment_request = None
+        error_message = None
+        try:
+            if data["error"]["message"]:
+                logger.error(data["error"]["message"])
+                error_message = data["error"]["message"]
+                return InvoiceResponse(
+                    False, checking_id, payment_request, error_message
+                )
+        except:
+            checking_id, payment_request = (
+                data["result"]["payment_hash"],
+                data["result"]["invoice"],
```

Comment:
Same here, this line might thrown an exception, because you don't know if these fields are present in the dictionary. Use `.get("key") is None` first to be sure, otherwise this will crash when cliche misbehaves and the exception won't be caught in lnbits.

Reply:
Done